### DISCUSSION TITLE
Codespaces didn't have my ssh keys and wasn't authorized to reach https github

### DISCUSSION THREAD

**Discussion by krainboltgreene** [community/community#9571] [Codespaces]
I created a personal codespace from a github repository I didn't own.

I wrote code.

I tried to push with the sync button. VSCode helpfully tells me I don't have permission to push to x/y, asks if I want to fork the repository?

I click yes.

It tells me I don't have permission to push to krainboltgreene/y, would I like to fork it? (This will probably be an infinite loop)

Turns out my vscode doesn't have my ssh keys, didn't have my dotfiles, and the https url VScode helpfully set as my `origin` repository just 403's without any input.

**Comment by shellerci** [comment]
Have you enabled access to the repository you're trying to push to (or all your repositories)?
https://docs.github.com/en/codespaces/managing-your-codespaces/managing-repository-access-for-your-codespaces

**Comment by krainboltgreene** [comment]
*After rereading your comment*

No I had not set that, it's a very scary option based on the description alone. All codespaces can access my repositories?

**Reply by krainboltgreene** [reply]
Further, based on the description it should currently work, yes? They were opened for my repository.

**Comment by kleenkanteen** [comment]
You can also add your private ssh key linked to your github account, onto your codespace. I just did that and was able to push to my fork. You also have to change the git origin with `git remote set-url origin new.git.url/here`. then start the ssh agent with `eval "$(ssh-agent -s)"`, create the ~/.ssh/id_rsa file, do `chmod 700 ~/.ssh/id_rsa` to make permissions only for the user, then `ssh-add ~/.ssh/id_rsa` to finally add the key so you can push to your fork. I just did all that and it worked.

**Reply by krainboltgreene** [reply]
This isn't a post about "How do I do X?", that's googable, this post is "The rubegoldbergian machine was not working as expected based on the documentation."

After many years I suspect that there was some conflict between the docker build, the home setup script, and github's own codespaces setup that created a bad scenario.

**Reply by kleenkanteen** [reply]
ok, well I came across this post on google. And I gave a workaround for others who might see this as well. Why so rude?

**Reply by JonasSonn** [reply]
> ok, well I came across this post on google. And I gave a workaround for others who might see this as well. Why so rude?

Thanks, this helped me at least!
